### PR TITLE
Add reusable workflows for linting

### PR DESCRIPTION
**Pull request by at-wat**
Reuse lint/format workflows for ROS package repositories.

https://docs.github.com/en/actions/using-workflows/reusing-workflows

### CODE REVIEW

File: .github/workflows/cpp.yml [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+        value: ${{ jobs.roslint.outputs.passed }}
+
+jobs:
+  clang-format:
+    runs-on: ubuntu-latest
+    outputs:
+      passed: ${{ steps.check.outputs.passed }}
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
```
Comment (f-fl0) [inline .github/workflows/cpp.yml:27]: We could use `actions/checkout@v3`

File: .github/workflows/cpp.yml [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+              | sed 's/^\([^:]\+\):\([0-9]\+\):\([0-9]\+\): error: \+\(.*\)/::error file=\1,line=\2,col=\3::\4/'
+          ) && echo "::set-output name=passed::true" \
+            || (echo "::set-output name=passed::false"; exit 1)
+  roslint:
+    runs-on: ubuntu-latest
+    outputs:
+      passed: ${{ steps.check.outputs.passed }}
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
```
Comment (f-fl0) [inline .github/workflows/cpp.yml:63]: Same here about `actions/checkout@v3`

File: .github/workflows/cpp.yml [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+
+          files_to_check="${{ inputs.files }}"
+          if [ -z ${files_to_check} ]; then
+            files_to_check=$(find . -name "*.cpp" -o -name "*.h")
+          fi
+
+          (
+            set -eo pipefail
+            roslint-cpplint \
+                --filter=-runtime/references,-build/c++11 \
```
Comment (f-fl0) [inline .github/workflows/cpp.yml:84]: Maybe the filter could be based on an argument or loaded from the file and default to `-runtime/references,-build/c++11`.

File: .github/workflows/cpp.yml [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+            set -eo pipefail
+            clang-format \
+                --style=file --dry-run --Werror \
+                ${files_to_check} \
+                2>&1 > /dev/null \
+              | sed ':l; N; /\n\S\+:[0-9]\+:[0-9]\+: error: /b l; N; s/\n/%0A/g' \
+              | sed 's/^\([^:]\+\):\([0-9]\+\):\([0-9]\+\): error: \+\(.*\)/::error file=\1,line=\2,col=\3::\4/'
+          ) && echo "::set-output name=passed::true" \
+            || (echo "::set-output name=passed::false"; exit 1)
+  roslint:
```
Comment (f-fl0) [inline .github/workflows/cpp.yml:57]: Maybe the `roslint` job could be separated as `roslint` is not checking only `c++` code.

Comment (at-wat) [reply .github/workflows/cpp.yml:57]: Does roslint-pep8 differ from normal pep8 contained in flake?

Comment (f-fl0) [reply .github/workflows/cpp.yml:60]: I thought `roslint` would also check things like the presence of copyright header in python modules/scripts but it doesn't. It is fine to leave as you wrote it.

**Review by f-fl0 — APPROVED** [review]
LGTM

**Review by Taka-Kazu — APPROVED** [review]

**Pull request merged**